File: ui/src/types/author-key.ts
export interface AuthorKeySpec {
  ownerName: string
  displayName: string
  fingerprint: string
  algorithm: string
  publicKey: string
  createdAt: string
}

export interface AuthorKeyMetadata {
  name: string
  version?: number
  creationTimestamp?: string
}

export interface AuthorKeyResource {
  apiVersion: 'privateposts.halo.run/v1alpha1'
  kind: 'AuthorKey'
  metadata: AuthorKeyMetadata
  spec: AuthorKeySpec
}

export interface AuthorKeyList {
  items?: AuthorKeyResource[]
  total?: number
}

export interface LocalAuthorKeyRecord {
  ownerName: string
  displayName: string
  fingerprint: string
  algorithm: string
  publicKey: JsonWebKey
  privateKey: JsonWebKey
  createdAt: string
}

export interface AuthorKeyRecipient {
  keyId: string
  algorithm: string
  publicKey: JsonWebKey
}

File: src/main/java/run/halo/privateposts/model/AuthorKey.java
package run.halo.privateposts.model;

import static io.swagger.v3.oas.annotations.media.Schema.RequiredMode.REQUIRED;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import run.halo.app.extension.AbstractExtension;
import run.halo.app.extension.GVK;

@Data
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
@GVK(
    group = "privateposts.halo.run",
    version = "v1alpha1",
    kind = "AuthorKey",
    singular = "authorkey",
    plural = "authorkeys"
)
public class AuthorKey extends AbstractExtension {
    @Schema(requiredMode = REQUIRED)
    private AuthorKeySpec spec;

    @Data
    public static class AuthorKeySpec {
        @Schema(requiredMode = REQUIRED, minLength = 1)
        private String ownerName;

        @Schema(requiredMode = REQUIRED, minLength = 1)
        private String displayName;

        @Schema(requiredMode = REQUIRED, minLength = 1)
        private String fingerprint;

        @Schema(requiredMode = REQUIRED, minLength = 1)
        private String algorithm;

        @Schema(requiredMode = REQUIRED, minLength = 1)
        private String publicKey;

        @Schema(requiredMode = REQUIRED, minLength = 1)
        private String createdAt;
    }
}
